lib/spack/spack/cmd/mirror.py

```diff
@@ -599,46 +599,49 @@ def process_mirror_stats(present, mirrored, error):
 
 def mirror_create(args):
     """create a directory to be used as a spack mirror, and fill it with package archives"""
-    if args.file and args.all:
-        raise SpackError(
-            "cannot specify specs with a file if you chose to mirror all specs with '--all'"
-        )
-
-    if args.file and args.specs:
-        raise SpackError("cannot specify specs with a file AND on command line")
-
-    if not args.specs and not args.file and not args.all:
-        raise SpackError(
-            "no packages were specified.",
-            "To mirror all packages, use the '--all' option "
-            "(this will require significant time and space).",
-        )
+    spack.mirrors.utils.set_mirror_all(bool(args.all))
+    try:
+        if args.file and args.all:
```

Comment:
It looks like the "old" contents of this method got placed in a `try/finally` block: can you

* keep the indentation
* rename this method to `_mirror_create`
* have `mirror_create` do

```
    spack.mirrors.utils.set_mirror_all(bool(args.all))
    try:
        _mirror_create(args)
    finally:
        spack.mirrors.utils.set_mirror_all(False)
```

Slightly more-idiomatic is to turn this into a context manager like:

```
@contextlib.contextmanager
def mirror_across_all_archs():
    try:
        spack.mirrors.utils.set_mirror_all(bool(args.all))
        yield
    finally:
        spack.mirrors.utils.set_mirror_all(bool(False))
```